camayoc/tests/qpc/cli/test_openshift.py

```diff
@@ -0,0 +1,120 @@
+"""Tests for OpenShift sources.
+
+:caseautomation: automated
+:casecomponent: cli
+:caseimportance: high
+:caselevel: integration
+:requirement: OCP
+:testtype: functional
+:upstream: yes
+"""
+import json
+import re
+import tarfile
+
+import pytest
+
+from camayoc import utils
+from camayoc.constants import TOKEN_INPUT
+from camayoc.tests.qpc.cli.utils import config_openshift
+from camayoc.tests.qpc.cli.utils import cred_add_and_check
+from camayoc.tests.qpc.cli.utils import report_download
+from camayoc.tests.qpc.cli.utils import scan_add_and_check
+from camayoc.tests.qpc.cli.utils import scan_job
+from camayoc.tests.qpc.cli.utils import scan_start
+from camayoc.tests.qpc.cli.utils import source_add_and_check
+from camayoc.tests.qpc.cli.utils import wait_for_scan
+
+
+def retrieve_report(scan_name, scan_job_id):
+    output_file = f"report-{scan_name}.tar.gz"
+    report_download({"scan-job": scan_job_id, "output-file": output_file})
+    details = deployments = None
+    with tarfile.open(output_file) as pkg:
+        for member in pkg.getmembers():
+            if member.name.endswith("details.json"):
+                data = pkg.extractfile(member).read()
+                details = json.loads(data)
+            if member.name.endswith("deployments.json"):
+                data = pkg.extractfile(member).read()
+                deployments = json.loads(data)
+    return details, deployments
+
+
+def validate_openshift_report(cluster, details, deployments):
+    assert details["sources"]
+    assert deployments["system_fingerprints"]
+    cluster_nodes = cluster["nodes"]
+    cluster_version = cluster["version"]
+    cluster_id = cluster["cluster_id"]
+    cluster_hosts = cluster["hosts"]
```

Comment:
I was a little bit confused with the nomenclature for "cluster_hosts" until reading the rest of the code.
I'd call those "node_names" or something like that to better reflect what they are.

Reply:
I've renamed to `cluster_nodes` and `nodes` in the configuration.